<launch>
  <arg name="ns" default="kinect"/>

  <arg name="host" default="localhost"/>
  <arg name="optionals" default="false"/>
  <arg name="frame" default="base_link"/>
  <arg name="filter" default="true"/>

  <arg name="queue_size" default="30"/>
  <arg name="timestamp" default="0.2"/>

  <arg name="localradius_r" default="3.0"/>

  <group ns="$(arg ns)">
  <node name="kinect_point_stream" pkg="linux_kinect"
        type="kinect_point_stream.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="frame" value="$(arg frame)"/>
     <param name="ns" value="$(arg ns)"/>
  </node>
  <node name="kinect_camerainfo_stream" pkg="linux_kinect"
        type="kinect_camerainfo_stream.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="frame" value="$(arg frame)"/>
     <param name="ns" value="$(arg ns)"/>
  </node>

  <node name="speech_recognition" pkg="linux_kinect"
        type="speech_recognition.py" output="screen">
     <param name="ip" value="$(arg host)"/>
  </node>
  <node name="audio_beam" pkg="linux_kinect"
        type="audio_beam.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="ns" value="$(arg ns)"/>
  </node>
  <node name="windows_tts" pkg="linux_kinect"
        type="tts.py" output="screen">
     <param name="ip" value="$(arg host)"/>
  </node>

  <node name="kinect_get_points" pkg="linux_kinect"
        type="kinect_get_3d.py" output="screen">
     <param name="ip" value="$(arg host)"/>
  </node>
  <node name="kinect_get_image" pkg="linux_kinect"
        type="kinect_get_image.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="frame" value="$(arg frame)"/>
  </node>

  <node name="windows_ocr" pkg="linux_kinect"
        type="ocr.py" output="screen">
     <param name="ip" value="$(arg host)"/>
  </node>

  <node name="audio_stream" pkg="linux_kinect"
        type="audio_stream.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="ns" value="$(arg ns)"/>
  </node>

  <node name="extract_rgb_from_depth" pkg="linux_kinect"
        type="extract_rgb_from_depth" output="screen"
        unless="$(arg filter)">
     <param name="ns" value="$(arg ns)"/>
     <param name="localradius_r" value="$(arg localradius_r)"/>
  </node>

  <group if="$(arg filter)">
    <node name="extract_rgb_from_depth" pkg="linux_kinect"
          type="extract_rgb_from_depth_filtered" output="screen">
       <param name="ns" value="$(arg ns)"/>
       <param name="queue_size" value="$(arg queue_size)"/>
       <param name="timestamp" value="$(arg timestamp)"/>
    </node>
  </group>

  <node name="kinect_image_stream" pkg="linux_kinect"
        type="kinect_image_stream.py" output="screen">
     <param name="ip" value="$(arg host)"/>
     <param name="frame" value="$(arg frame)"/>
     <param name="ns" value="$(arg ns)"/>
  </node>
  </group>

</launch>